controllers/goflowkube/goflowkube_reconciler.go

```diff
@@ -63,6 +69,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, desired *flowsv1alpha1.FlowC
 			newSVC := buildService(desired, r.OperatorNamespace)
 			r.createOrUpdate(ctx, oldSVC, newSVC, serviceKind)
 		}
+
+		//Delete or Create / Update Autoscaler according to HPA option
+		if desired.HPA == nil {
+			r.delete(ctx, oldASC, autoscalerKind)
```

Comment:
what if `oldASC` is nil? We probably shouldn't try to delete it then

Reply:
also changed in daemonSetKind case for oldSVC